File: M3Tools/Controls/People/StateSelectorField.vb
Public Class StateSelectorField
    Public Property StateCode As String
        Get
			Return cbx_States.Text
		End Get
        Set(value As String)
			If value?.Length >= 2 Then
				value = value.Substring(0, 2)
			End If

			cbx_States.Text = value
		End Set
    End Property

    Private Sub cbx_States_TextChanged(sender As Object, e As EventArgs) Handles cbx_States.TextChanged

    End Sub
End Class

File: M3Tools/Controls/People/StateSelectorField.cs
using System;

namespace SPPBC.M3Tools
{
	/// <summary>
	/// 
	/// </summary>
    public partial class StateSelectorField
    {
		/// <summary>
		/// The selected state code
		/// </summary>
        public string StateCode
        {
            get
            {
                return cbx_States.Text;
            }
            set
            {
                if ((value?.Length) is { } arg1 && arg1 >= 2)
                {
                    value = value.Substring(0, 2);
                }

                cbx_States.Text = value;
            }
        }

		/// <summary>
		/// 
		/// </summary>
        public StateSelectorField()
        {
            InitializeComponent();
        }

        private void cbx_States_TextChanged(object sender, EventArgs e)
        {

        }
    }
}
